pkg/controller/master/node/cpu_manager_controller.go

```diff
@@ -0,0 +1,358 @@
+		return h.nodeClient.Update(updateNode(node, toFailedStatus(cpuManagerStatus)))
+	}
+
+	return h.nodeClient.Update(updateNode(node, toRunningStatus(cpuManagerStatus, job)))
+}
+
+func (h *cpuManagerNodeHandler) OnJobChanged(_ string, job *batchv1.Job) (*batchv1.Job, error) {
+	if job == nil || job.DeletionTimestamp != nil {
+		return job, nil
+	}
+
+	nodeName, ok := job.Labels[util.LabelCPUManagerUpdateNode]
+	if !ok {
+		return job, nil
+	}
+
+	node, err := h.nodeCache.Get(nodeName)
+	if err != nil {
+		return job, err
+	}
+
+	if condition.Cond(batchv1.JobComplete).IsTrue(job) {
+		updateStatus := &CPUManagerUpdateStatus{
+			Status:  CPUManagerSuccessStatus,
+			Policy:  CPUManagerPolicy(job.Labels[util.LabelCPUManagerUpdatePolicy]),
+			JobName: job.Name,
+		}
+		if _, err := h.nodeClient.Update(updateNode(node, updateStatus)); err != nil {
+			return nil, err
+		}
+	}
+
+	if condition.Cond(batchv1.JobFailed).IsTrue(job) {
+		updateStatus := &CPUManagerUpdateStatus{
+			Status:  CPUManagerFailedStatus,
+			Policy:  CPUManagerPolicy(job.Labels[util.LabelCPUManagerUpdatePolicy]),
+			JobName: job.Name,
+		}
+		if _, err := h.nodeClient.Update(updateNode(node, updateStatus)); err != nil {
+			return nil, err
+		}
+	}
+
+	return job, nil
+}
+
+func GetCPUManagerUpdateStatus(jsonString string) (*CPUManagerUpdateStatus, error) {
+	cpuManagerStatus := &CPUManagerUpdateStatus{}
+
+	cpuManagerStatus.Policy = "foo"
+
+	if err := json.Unmarshal([]byte(jsonString), cpuManagerStatus); err != nil {
+		return nil, err
+	}
+	if cpuManagerStatus.Status != CPUManagerRequestedStatus && cpuManagerStatus.Status != CPUManagerRunningStatus && cpuManagerStatus.Status != CPUManagerSuccessStatus && cpuManagerStatus.Status != CPUManagerFailedStatus {
+		return nil, errors.New("invalid status")
+	}
+	if cpuManagerStatus.Policy != CPUManagerNonePolicy && cpuManagerStatus.Policy != CPUManagerStaticPolicy {
+		return nil, errors.New("invalid policy")
+	}
+	return cpuManagerStatus, nil
+}
+
+func toFailedStatus(updateStatus *CPUManagerUpdateStatus) *CPUManagerUpdateStatus {
+	newUpdateStatus := CPUManagerUpdateStatus(*updateStatus)
+	newUpdateStatus.Status = CPUManagerFailedStatus
+	return &newUpdateStatus
+}
+
+func toRunningStatus(updateStatus *CPUManagerUpdateStatus, job *batchv1.Job) *CPUManagerUpdateStatus {
+	newUpdateStatus := CPUManagerUpdateStatus(*updateStatus)
+	newUpdateStatus.Status = CPUManagerRunningStatus
+	newUpdateStatus.JobName = job.Name
+	return &newUpdateStatus
+}
+
+func updateNode(node *corev1.Node, updateStatus *CPUManagerUpdateStatus) *corev1.Node {
+	jsonStr, _ := json.Marshal(updateStatus)
+	toUpdate := node.DeepCopy()
+	toUpdate.Annotations[util.AnnotationCPUManagerUpdateStatus] = string(jsonStr)
+	return toUpdate
+}
+
+func getScript(nodeName string, policy CPUManagerPolicy) string {
+	var label string
+	if policy == CPUManagerStaticPolicy {
+		label = "true"
+	} else {
+		label = "false"
+	}
+	return fmt.Sprintf(`
```

Comment:
Alternatively, you can store the script in a config map and mount the conifg map to the job. It provides some flexibility. We do something similar in the promote job (https://github.com/harvester/harvester/blob/7eb6121940c35b6172f6698a5081d314209d427e/deploy/charts/harvester/templates/configmap.yaml#L21)

Reply:
agree, we should store this in a configmap as it makes it easier to provide workarounds to handle edge cases